test/test.js

```diff
@@ -12,29 +12,56 @@ describe('Chess', function() {
     it('should initialize a game between two players', function (done) {
       this.timeout(10000);
 
-      Chess.initGame(player2, {from: player1, gas: 1000000});
+      Chess.initGame('Alice', {from: player1, gas: 1000000});
 
       var filter = Chess.GameInitialized({});
       filter.watch(function(error, result){
         gameId = result.args.gameId;
         assert.isOk(result.args.gameId);
+        assert.equal('Alice', result.args.player1Alias);
         filter.stopWatching();
         done();
       });
     });
   });
 
+  describe('joinGame()', function () {
+    it('should join player2', function(done) {
+      this.timeout(10000);
+
+      assert.doesNotThrow(function(){
+        Chess.joinGame(gameId, 'Bob', {from: player2, gas: 100000});
+      }, Error);
+
+      var filter = Chess.GameJoined({});
+      filter.watch(function(error, result){
+        assert.equal(gameId, result.args.gameId);
+        assert.equal(player2, result.args.player2);
+        assert.equal('Bob', result.args.player2Alias);
+        filter.stopWatching();
+        done();
+      });
+    });
+    it('should disallow joing a game with two payers', function() {
+      this.timeout(10000);
+
+      assert.throws(function(){
+        Chess.joinGame(gameId, 'Bob', {from: player2, gas: 100000});
```

Comment:
Du hast recht, das ist nicht ganz sauber, funktioniert aber, weil alle `it` hintereinander ausgeführt werden. Ich denke das ist an dieser Stelle ok, sonst würde man das doppelt testen. Allgemein hast du recht.